src/swap/SwapReviewScreen.tsx

```diff
@@ -0,0 +1,331 @@
+import { StackScreenProps } from '@react-navigation/stack'
+import BigNumber from 'bignumber.js'
+import qs from 'qs'
+import React, { useEffect, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import { PixelRatio, ScrollView, StyleSheet, Text, TouchableOpacity, View } from 'react-native'
+import { SafeAreaView } from 'react-native-safe-area-context'
+import { useDispatch, useSelector } from 'react-redux'
+import { showError } from 'src/alert/actions'
+import { SwapEvents } from 'src/analytics/Events'
+import ValoraAnalytics from 'src/analytics/ValoraAnalytics'
+import { ErrorMessages } from 'src/app/ErrorMessages'
+import { maxSwapSlippagePercentageSelector } from 'src/app/selectors'
+import BackButton from 'src/components/BackButton'
+import Button, { BtnSizes } from 'src/components/Button'
+import Dialog from 'src/components/Dialog'
+import CustomHeader from 'src/components/header/CustomHeader'
+import TokenDisplay, { formatValueToDisplay } from 'src/components/TokenDisplay'
+import InfoIcon from 'src/icons/InfoIcon'
+import LoadingSpinner from 'src/icons/LoadingSpinner'
+import Times from 'src/icons/Times'
+import { noHeader } from 'src/navigator/Headers'
+import { navigateBack, navigateHome } from 'src/navigator/NavigationService'
+import { Screens } from 'src/navigator/Screens'
+import { TopBarIconButton, TopBarTextButton } from 'src/navigator/TopBarButton'
+import { StackParamList } from 'src/navigator/types'
+import DisconnectBanner from 'src/shared/DisconnectBanner'
+import colors from 'src/styles/colors'
+import fontStyles from 'src/styles/fonts'
+import { Spacing } from 'src/styles/styles'
+import variables from 'src/styles/variables'
+import { Field } from 'src/swap/useSwapQuote'
+import { coreTokensSelector } from 'src/tokens/selectors'
+import { divideByWei, multiplyByWei } from 'src/utils/formatting'
+import Logger from 'src/utils/Logger'
+import networkConfig from 'src/web3/networkConfig'
+import { accountAddressSelector } from 'src/web3/selectors'
+
+type Props = StackScreenProps<StackParamList, Screens.SwapReviewScreen>
+
+const TAG = 'SWAP_REVIEW_SCREEN'
+
+export function SwapReviewScreen(props: Props) {
+  const { toToken, fromToken, swapAmount, updatedField } = props.route.params
+  const [swapInfo, setSwapInfo] = useState(null as any)
+  const [loading, setLoading] = useState(false)
+  const [fetchSwapQuoteError, setFetchSwapQuoteError] = useState(false)
+  const [estimatedModalVisible, setEstimatedDialogVisible] = useState(false)
+  const coreTokens = useSelector(coreTokensSelector)
+  const maxSlippagePercent = useSelector(maxSwapSlippagePercentageSelector)
+  const walletAddress = useSelector(accountAddressSelector)
```

Comment:
this should be `walletAddressSelector` instead - we don't want the mtw address here.

Reply:
fixed in https://github.com/valora-inc/wallet/pull/2767/commits/90d9446a7b9a6a9369bfc81b9d2ca3a181447742!